### PR TITLE
docs: clarifications around external outputs info.

### PR DESCRIPTION
**Pull request by jorgeorpinel**
- Followup to #2104
- ~~Rename "Managing External Data" guide to "External Outputs"~~
- Mentions of `add/import-url --to-cache/remote`

### CODE REVIEW

**Comment by shcheklein** [comment]
@jorgeorpinel let's postpone this please. It doesn't affect 2.0 and we can review/merge it after I think.

File: content/docs/user-guide/external-outputs.md [hunk 1]
```diff
@@ -1,51 +1,48 @@
...
+# External Outputs
 
-> ⚠️ This is an advanced feature that we don't recommend using unless you really
-> know what you are doing. Artifacts added with --external are not affected by
-> `dvc push/pull/status -c`. You are likely looking for
-> [straight-to-remote/cache](https://github.com/iterative/dvc/issues/4520)
-> functionality or `dvc import-url`
+> ⚠️ This is an advanced feature for very specific situations and not
+> recommended except if there's absolutely no other alternative. In most cases
+> alternatives like the `--to-cache` or `--to-remote` options of `dvc add` and
```
Comment (jorgeorpinel) [inline content/docs/user-guide/external-outputs.md:5]:
```suggestion
> alternatives like the `--to-cache` or `--to-remote` options of `dvc add` and
```

These can link to ref examples now.

File: content/docs/user-guide/external-outputs.md [hunk 2]
```diff
@@ -1,51 +1,48 @@
-# Managing External Data
+# External Outputs
```
Comment (jorgeorpinel) [inline content/docs/user-guide/external-outputs.md:1]: Apparently this name was quite intentional as "outputs" is not meaningful for end-users (at the user guide level)...

Comment (jorgeorpinel) [reply content/docs/user-guide/external-outputs.md]: Reverted the name. Let's just try to merge the copy edits then? Thanks!

**Comment by jorgeorpinel** [comment]
I reverted the guide name change in 49eefb0 @shcheklein. PTAL when you have a chance.

**Review by shcheklein — APPROVED** [review]

**Pull request merged**